supervisor/utils/pathlib.py

```diff
@@ -0,0 +1,18 @@
+"""Pathlib utility for exclusion filtering."""
+import logging
+from pathlib import PurePath
+from typing import List
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+def is_excluded_by_filter(path: PurePath, exclude_list: List[str]) -> bool:
```

Comment:
Let's keep it more simple, we use that only inside tar utils. Just copy the function back to tar utils as `_is_exlude_by_filter` and use it inside the function.